src/test/groovy/com/github/rodionovsasha/commentservice/controllers/UserControllerTest.groovy

```diff
@@ -0,0 +1,88 @@
+package com.github.rodionovsasha.commentservice.controllers
+
+import com.github.rodionovsasha.commentservice.exceptions.InactiveUserException
+import org.springframework.http.HttpStatus
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get
+import com.github.rodionovsasha.commentservice.entities.User
+import com.github.rodionovsasha.commentservice.exceptions.UserNotFoundException
+import com.github.rodionovsasha.commentservice.services.UserService
+import spock.lang.Specification
+
+import static com.github.rodionovsasha.commentservice.Application.API_BASE_URL
+import static org.springframework.util.MimeTypeUtils.APPLICATION_JSON_VALUE
+import static org.springframework.http.MediaType.APPLICATION_JSON_UTF8_VALUE
+import static org.springframework.test.web.servlet.setup.MockMvcBuilders.standaloneSetup
+
+class UserControllerTest extends Specification {
+    final HOMER_ID = 1
+    final GET_HOMER_URL = API_BASE_URL + "/user/1"
+
+    def service = Mock(UserService)
+    def controller = new UserController(service)
+    def user = new User()
+
+    def mockMvc = standaloneSetup(controller).setControllerAdvice(new ExceptionHandlerController()).build()
+
+    def setup() {
+        user.id = HOMER_ID
+        user.name = "Homer"
+        user.age = 39
+        user.active
+    }
+
+    def "should get active user"() {
+        when:
+        def response = mockMvc.perform(get(GET_HOMER_URL).contentType(APPLICATION_JSON_VALUE)).andReturn().response
+
+        then:
+        1 * service.getActiveUser(HOMER_ID) >> user
+        with(response) {
+            status == HttpStatus.OK.value()
+            contentType == APPLICATION_JSON_UTF8_VALUE
+            contentAsString == '{"id":1,"name":"Homer","age":39,"active":true}'
```

Comment:
I would rather compare JSON structures instead of strings:
```groovy
new JsonSlurper().parseText(contentAsString) == [id: 1, name: "Homer", age: 39, active: true]
```

Reply:
:+1: